config.h

```diff
@@ -1,216 +1,172 @@
+  { (ControlMask | ShiftMask),  XK_C,           clipcopy,       {.i =  0} },
+  { (ControlMask | ShiftMask),  XK_V,           clippaste,      {.i =  0} },
+  { (ControlMask | ShiftMask),  XK_Y,           selpaste,       {.i =  0} },
+  { (ControlMask | ShiftMask),  XK_Num_Lock,    numlock,        {.i =  0} },
+  { (ControlMask | ShiftMask),  XK_I,           iso14755,       {.i =  0} },
 };
 
-/*
- * Special keys (change & recompile st.info accordingly)
- *
- * Mask value:
- * * Use XK_ANY_MOD to match the key no matter modifiers state
- * * Use XK_NO_MOD to match the key alone (no modifiers)
- * appkey value:
- * * 0: no value
- * * > 0: keypad application mode enabled
- * *   = 2: term.numlock = 1
- * * < 0: keypad application mode disabled
- * appcursor value:
- * * 0: no value
- * * > 0: cursor application mode enabled
- * * < 0: cursor application mode disabled
- * crlf value
- * * 0: no value
- * * > 0: crlf mode is enabled
- * * < 0: crlf mode is disabled
- *
- * Be careful with the order of the definitions because st searches in
- * this table sequentially, so any XK_ANY_MOD must be in the last
- * position for a key.
- */
-
-/*
- * If you want keys other than the X11 function keys (0xFD00 - 0xFFFF)
- * to be mapped below, add them to this array.
- */
+// Keys outside the X11 function key range (0xFD00 - 0xFFFF) with shortcuts.
+// By default, shortcuts in `key` are only scanned for KeySyms in this range.
 static KeySym mappedkeys[] = {-1};
 
-/*
- * State bits to ignore when matching key or button events.  By default,
- * numlock (Mod2Mask) and keyboard layout (XK_SWITCH_MOD) are ignored.
- */
+// Modifiers to ignore when matching keyboard and mouse events.
+// By default, numlock and keyboard layout are ignored.
 static uint ignoremod = Mod2Mask | XK_SWITCH_MOD;
 
-/*
- * Override mouse-select while mask is active (when MODE_MOUSE is set).
- * Note that if you want to use ShiftMask with selmasks, set this to an other
- * modifier, set to 0 to not use it.
- */
+// Keyboard modifier to force mouse selection when in mouse mode.
+// This lets you select text even if the application has captured the mouse.
 uint forceselmod = ShiftMask;
 
-/*
- * This is the huge key array which defines all compatibility to the Linux
- * world. Please decide about changes wisely.
- */
+// Keyboard shortcuts that send bytes to the terminal.
+// The first entry whose critera match the key event is chosen.
+//
+// Some criteria match against the current mode:
+//   `appkey` is linked to whether the keypad is in "application mode"
+//   `appcursor` is linked to whether cursor keys are in "application mode"
+//   `crlf` is linked to whether the terminal is in "automatic newline mode"
+// These have three values
+//   -1: only match if mode is NOT active
+//   0 : match independent of mode
+//   1 : only match if mode is active
+// appkey has one extra value:
+//   2 : only match if mode is active AND numlock was toggled off.
```

Comment:
That's what the old comment said, but in the `kmap()` loop:
```
if (term.numlock && kp->appkey == 2)
      continue;
```
So if `numlock` is 1, then we *skip* rules with `appkey==2`

Maybe `numlock=1` means *off* and 0 means *on*?

Looking at where this was added in 172f65436ce14a52842d67b862fdc45f8ff3ada3, my interpretation is:
  - this changed the default from generating dedicated escape sequences for `XK_KP_2` and friends, to just generating `2`
  - it seems reasonable to call the new mode numlock, and indeed it was indicated by `numlock=1`
  - so the old comment is wrong and the new one is correct
  - but I have very little confidence the behavior is actually good, worth checking what other terms, apps, tmux think about these keys